esxi/tests/test_unit.py

```diff
@@ -20,10 +20,11 @@ def test_esxi_metric_up(instance, dd_run_check, aggregator, caplog):
     assert "Connected to ESXi host localhost: VMware ESXi 6.5.0 build-123456789" in caplog.text
 
 
-def test_emits_critical_service_check_when_service_is_down(dd_run_check, aggregator, instance, caplog):
+def test_emits_critical_service_check_when_service_is_down(dd_run_check, aggregator, caplog, instance):
```

Comment:
I'm afraid that this is more of an integration test than a unit test, since we are not mocking or patching the connection at any time. We simply have an error because there is no process listening on the configured ip and port.